cli/util/uploader.go

```diff
@@ -0,0 +1,360 @@
+					Scheme: "neofs",
+					Opaque: fmt.Sprintf("%s/%s/header", containerID.String(), currOid.String()),
+				}
+				headerRes, err := neofs.GetWithClient(ctxWithCancel, clientSDK, account.PrivateKey(), url, false)
+				if err != nil {
+					errCh <- fmt.Errorf("failed to get header for OID %s: %w", currOid.String(), err)
+					cancel()
+					return
+				}
+				defer headerRes.Close()
+
+				headerBytes, err := gio.ReadAll(headerRes)
+				if err != nil {
+					errCh <- fmt.Errorf("failed to read header for OID %s: %w", currOid.String(), err)
+					cancel()
+					return
+				}
+
+				blockIndex, err := processHeader(headerBytes, attributeKey)
+				if err != nil {
+					errCh <- fmt.Errorf("failed to process header for OID %s: %w", currOid.String(), err)
+					cancel()
+					return
+				}
+
+				offset := blockIndex * oidSize
+				res := make([]byte, oidSize)
+				currOid.Encode(res)
+				copy(buffer[offset:], res)
+			}(currOid)
+		}
+		wg.Wait()
+
+		select {
+		case err := <-errCh:
+			return err
+		default:
+		}
+		attrs := []object.Attribute{
+			*object.NewAttribute(attributeKey, strconv.Itoa(int(i))),
+			*object.NewAttribute("size", strconv.Itoa(int(indexFileSize))),
+		}
+		err = uploadBlock(ctx.Context, p, user.NewAutoIDSignerRFC6979(account.PrivateKey().PrivateKey), account.PrivateKey().GetScriptHash(), containerID, buf.Bytes(), attrs)
+		if err != nil {
+			return fmt.Errorf("failed to upload index file %d: %w", i, err)
+		}
+		fmt.Fprintf(ctx.App.Writer, "Uploaded index file %d\n", i)
+		buf.Reset()
+	}
+
+	return nil
+}
+
+// uploadBlock uploads the block to the container using the pool.
+func uploadBlock(ctx context.Context, p pool.Pool, signer user.Signer, owner util.Uint160, containerID cid.ID, objData []byte, attrs []object.Attribute) error {
+	var ownerID user.ID
+	ownerID.SetScriptHash(owner)
+
+	var hdr object.Object
+	hdr.SetContainerID(containerID)
+	hdr.SetOwnerID(&ownerID)
+	hdr.SetAttributes(attrs...)
+
+	var prmObjectPutInit client.PrmObjectPutInit
+
+	writer, err := p.ObjectPutInit(ctx, hdr, signer, prmObjectPutInit)
+	if err != nil {
+		return fmt.Errorf("failed to initiate object upload: %w", err)
+	}
+	_, err = writer.Write(objData)
+	if err != nil {
+		return fmt.Errorf("failed to write object data: %w", err)
+	}
+	err = writer.Close()
+	if err != nil {
+		return fmt.Errorf("failed to finalize object upload: %w", err)
+	}
+	return nil
+}
+
+type attribute struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type customHeader struct {
+	Attributes []attribute `json:"attributes"`
+}
+
+func processHeader(headerBytes []byte, attribute string) (int, error) {
+	var header customHeader
+
+	// var obj object.Object
+	//err := obj.UnmarshalJSON(headerBytes) -  failed to unmarshal header: proto: (line 1:2): unknown field "version"
+	//err := obj.Unmarshal(headerBytes) - failed to unmarshal header: proto: cannot parse invalid wire-format data
+	if err := json.Unmarshal(headerBytes, &header); err != nil {
```

Comment:
BTW, why JSON is used to return headers? Does SDK return a binary format of headers?